src/pages/CookieClicker.vue

```diff
@@ -2,48 +2,186 @@
 import { computed, ref } from 'vue';
 
 const cookies = ref(0);
+
 const buildings = ref([
-    { name: 'Cursor', price: 15, cps: 0.1, count: 0},
-    { name: 'Grandma', price: 100, cps: 1, count: 0},
-    { name: 'Farm', price: 1000, cps: 10, count: 0},
-    { name: 'Factory', price: 10_000, cps: 100, count: 0},
+    { name: 'Cursor', price: 15, cps: 0.1, count: 0 },
+    { name: 'Grandma', price: 100, cps: 1, count: 0 },
+    { name: 'Farm', price: 1000, cps: 10, count: 0 },
+    { name: 'Factory', price: 10_000, cps: 100, count: 0 },
 ]);
 
-function buyBuilding(building){
-    cookies.value -= building.price;
-    building.price += Math.ceil(building.price / 100 * 15);
-    building.count++;
+
+const upgrades = ref([
+    { name: 'Better Ovens', effect: () => multiplyCps(2), price: 500, bought: false },
+    { name: 'Fertilized Fields', effect: () => multiplyCps(2), price: 5000, bought: false },
+]);
+
+// Achievements list
+const achievements = ref([]);
+const achieved = ref([]);
+
+// Shiny cookie state
+const shinyCookie = ref({ visible: false, x: 0, y: 0 });
+
+// Function to buy a building
+function buyBuilding(building) {
+    if (cookies.value >= building.price) {
+        cookies.value -= building.price;
+        building.price = Math.ceil(building.price * 1.15);
+        building.count++;
+        checkAchievements();
+    }
+}
+
+// Function to buy an upgrade
+function buyUpgrade(upgrade) {
+    if (cookies.value >= upgrade.price && !upgrade.bought) {
+        cookies.value -= upgrade.price;
+        upgrade.effect();
+        upgrade.bought = true;
+    }
 }
+
+// Function to multiply CPS (used for upgrades)
+function multiplyCps(factor) {
+    buildings.value.forEach(building => {
+        building.cps *= factor;
+    });
+}
+
+// Computed total CPS
 let cps = computed(() => {
-    let cps = 0;
+    let totalCps = 0;
     buildings.value.forEach(building => {
-       cps+=building.cps*building.count;
+        totalCps += building.cps * building.count;
     });
-    return cps;
+    return totalCps;
 });
-setInterval(()=>{
-   cookies.value+=cps.value;
 
-   document.title ='🍪' + +cookies.value.toFixed(1) + ' Cookies!';
-},1000);
+// Auto-generate cookies based on CPS every second
+setInterval(() => {
+    cookies.value += cps.value;
+    document.title = '🍪 ' + cookies.value.toFixed(1) + ' Cookies!';
+}, 1000);
 
+// Function to check achievements
+function checkAchievements() {
+    const newAchievements = [
+        { name: "100 Cookies!", condition: () => cookies.value >= 100 },
```

Comment:
so if I buy a some building and then wait will I ever get this achievment if I never buy another building?